jupyter_client/kernelapp.py

```
@@ -2,87 +2,92 @@
-            self.loop.add_callback_from_signal(self.shutdown, signo)
+            def inner_callback():
+                self.shutdown(signo)
+            self.loop.add_callback_from_signal(inner_callback)
 
         for sig in [signal.SIGTERM, signal.SIGINT]:
             signal.signal(sig, shutdown_handler)
-
-    def shutdown(self, signo: int) -> None:
+            
+    def shutdown(self, signo: int, unused_param=None) -> None:
+        l = []  
```

Ruff (0.8.2)

Ambiguous variable name: `l`

(E741)